internal/controllers/provisioningrequest_controller.go

```diff
@@ -1291,6 +1363,13 @@ func (t *provisioningRequestReconcilerTask) updateConfigurationAppliedStatus(
 			"error determining if the cluster is ready for policy configuration: %w", err)
 	}
 
+	if utils.IsClusterProvisionCompleted(t.object) &&
```

Comment:
It looks like this can be moved under the `!clusterIsReadyForPolicyConfig` below as
```
if utils.IsClusterProvisionCompleted(t.object) && utils.IsClusterProvisionCompleted(t.object) {
...
```